src/worker/EnigmaNode.js

```diff
@@ -54,25 +55,29 @@ class EnigmaNode extends EventEmitter {
    * @param {Function} callback - empty callback with no params
    */
   loadNode(path, callback) {
-    PeerId.createFromJSON(require(path), (err, idListener) => {
-      if (err) throw err;
+    fs.readFile(path, (err, data) => {
+      if (err) callback(err);
 
-      const peerInfo = new PeerInfo(idListener);
-      this.multiAddrs.forEach(addr => {
-        peerInfo.multiaddrs.add(addr);
-      });
-      this.node = new PeerBundle({
-        peerInfo,
-        config: {
-          peerDiscovery: {
-            bootstrap: {
-              enabled: this.isDiscover,
-              list: this.dnsNodes
+      PeerId.createFromJSON(JSON.parse(data), (err, idListener) => {
+        if (err) callback(err);
+
+        const peerInfo = new PeerInfo(idListener);
+        this.multiAddrs.forEach(addr => {
+          peerInfo.multiaddrs.add(addr);
+        });
+        this.node = new PeerBundle({
+          peerInfo,
+          config: {
+            peerDiscovery: {
+              bootstrap: {
+                enabled: this.isDiscover,
+                list: this.dnsNodes
+              }
             }
           }
-        }
+        });
+        setTimeout(callback, 100);
       });
-      setTimeout(callback, 100);
     });
   }
```

Comment:
Please use `util.promisify(fs.readFile)(path)` or `require(path)`.
`require` Is okay here since the file is small. It will also do `JSON.parse` for you.

Reply:
The reason I've removed the `require` is in order to enable specifying json files without the `json`extension (as was requested by @Cashmaney )